### PR TITLE
Non-GUID link updates

### PR DESCRIPTION
**Pull request by guardrex**
Addresses Part 3(2) of #2580

Converts Non-GUID MSDN links ...

```
https?://msdn.microsoft.com(/en-us)?/library/[A-Za-z]{2}[\d]{6}\(v=vs\.[\d]{2,3}\).aspx
```

All link changes are logged. No bad links were found on this pass.

[log-nonGUID.txt](https://github.com/dotnet/docs/files/1150462/log-nonGUID.txt) (CSV file)

### CODE REVIEW

File: docs/framework/migration-guide/net-framework-4-migration-issues.md [hunk 1]
```diff
@@ -296,7 +296,7 @@ Namespaces: <xref:System.Xml.Linq>; <xref:System.Xml.Schema>, <xref:System.Xml.X
 | **Namespace attribute** | To prevent data corruption, an <xref:System.Xml.XPath.XPathNavigator> object now returns the local name of the `x:xmlns` attribute correctly. |
 | **Namespace declarations** | An <xref:System.Xml.XmlReader> object on a sub-tree no longer creates duplicate namespace declarations within one XML element. |
 | **Schema validation** | To prevent erroneous schema validation, the <xref:System.Xml.Schema.XmlSchemaSet> class allows for XSD schemas to be compiled correctly and consistently. These schemas can include other schemas; for example, `A.xsd` can include `B.xsd`, which can include `C.xsd`. Compiling any one of these causes this graph of dependencies to be traversed. |
-| **Script functions** | The [function-available function](https://msdn.microsoft.com/library/ms256124(v=vs.100).aspx) no longer incorrectly returns `false` when the function is actually available. |
+| **Script functions** | The [function-available function](https://msdn.microsoft.com/library/ms256124(v=vs.110).aspx) no longer incorrectly returns `false` when the function is actually available. |
```
Comment (mairaw) [inline docs/framework/migration-guide/net-framework-4-migration-issues.md]: this is odd. it seems this content should be redirecting to docs but it's not. can you please open an issue for me to investigate this?

File: docs/framework/migration-guide/net-framework-4-migration-issues.md [hunk 2]
```diff
@@ -321,16 +321,16 @@ Namespaces: <xref:System.Xml.Linq>; <xref:System.Xml.Schema>, <xref:System.Xml.X
 
 ### Reference
 
-[New Types and Members in the .NET Framework 4](https://msdn.microsoft.com/library/ff641764(v=vs.100).aspx)
+[New Types and Members in the .NET Framework 4](https://msdn.microsoft.com/library/ff641764(v=vs.110).aspx)
```
Comment (mairaw) [inline docs/framework/migration-guide/net-framework-4-migration-issues.md]: vs.100 was the correct version

File: docs/framework/migration-guide/net-framework-4-migration-issues.md [hunk 3]
```diff
@@ -321,16 +321,16 @@ Namespaces: <xref:System.Xml.Linq>; <xref:System.Xml.Schema>, <xref:System.Xml.X
...
 
 ### Concepts
 
-[Migration Guide to the .NET Framework 4](https://msdn.microsoft.com/library/ff657133(v=vs.100).aspx)   
-[What's New in the .NET Framework 4](https://msdn.microsoft.com/library/ms171868(v=vs.100).aspx)   
+[Migration Guide to the .NET Framework 4](https://msdn.microsoft.com/library/ff657133(v=vs.110).aspx)
```
Comment (mairaw) [inline docs/framework/migration-guide/net-framework-4-migration-issues.md]: same for these other links

File: docs/framework/wpf/advanced/cleartype-registry-settings.md [hunk 4]
```diff
@@ -122,4 +122,4 @@ This topic provides an overview of the              [!INCLUDE[TLA2#tla_winclient
   
 ## See Also  
  [ClearType Overview](../../../../docs/framework/wpf/advanced/cleartype-overview.md)   
- [ClearType Antialiasing](https://msdn.microsoft.com/library/dd183433(v=vs.85).aspx)
\ No newline at end of file
+ [ClearType Antialiasing](https://msdn.microsoft.com/library/windows/desktop/dd183433(v=vs.85).aspx)
```
Comment (mairaw) [inline docs/framework/wpf/advanced/cleartype-registry-settings.md]: previous link was correct

File: docs/standard/microservices-architecture/multi-container-microservice-net-applications/subscribe-events.md [hunk 5]
```diff
@@ -81,11 +81,11 @@ In this case, since the origin microservice is a simple CRUD microservice, that
...
 
-In microservices-based architectures, you should choose availability and tolerance, and you should deemphasize strong consistency. Therefore, in most modern microservice-based applications, you usually do not want to use distributed transactions in messaging, as you do when you implement [distributed transactions](https://msdn.microsoft.com/en-us/library/ms978430.aspx#bdadotnetasync2_topic3c) based on the Windows Distributed Transaction Coordinator (DTC) with [MSMQ](https://msdn.microsoft.com/en-us/library/ms711472(v=vs.85).aspx).
+In microservices-based architectures, you should choose availability and tolerance, and you should deemphasize strong consistency. Therefore, in most modern microservice-based applications, you usually do not want to use distributed transactions in messaging, as you do when you implement [distributed transactions](https://msdn.microsoft.com/en-us/library/ms978430.aspx#bdadotnetasync2_topic3c) based on the Windows Distributed Transaction Coordinator (DTC) with [MSMQ](https://msdn.microsoft.com/library/ms711472(v=vs.85).aspx).
```
Comment (mairaw) [inline docs/standard/microservices-architecture/multi-container-microservice-net-applications/subscribe-events.md]: there's a link with en-us still

File: docs/visual-basic/programming-guide/concepts/async/index.md [hunk 6]
```diff
@@ -153,7 +153,7 @@ Dim urlContents As String = Await client.GetStringAsync()
...
   
- The Windows Runtime also contains many methods that you can use with `Async` and `Await` in Windows apps. For more information and example methods, see [Quickstart: using the await operator for asynchronous programming](http://go.microsoft.com/fwlink/?LinkId=248545), [Asynchronous programming (Windows Store apps)](http://go.microsoft.com/fwlink/?LinkId=259592), and [WhenAny: Bridging between the .NET Framework and the Windows Runtime](https://msdn.microsoft.com/library/jj635140(v=vs.120).aspx).  
+ The Windows Runtime also contains many methods that you can use with `Async` and `Await` in Windows apps. For more information and example methods, see [Quickstart: using the await operator for asynchronous programming](http://go.microsoft.com/fwlink/?LinkId=248545), [Asynchronous programming (Windows Store apps)](http://go.microsoft.com/fwlink/?LinkId=259592), and [WhenAny: Bridging between the .NET Framework and the Windows Runtime](https://msdn.microsoft.com/library/jj635140.aspx).
```
Comment (mairaw) [inline docs/visual-basic/programming-guide/concepts/async/index.md]: keep the old link

File: docs/visual-basic/programming-guide/concepts/async/index.md [hunk 7]
```diff
@@ -256,7 +256,7 @@ Await Task_MethodAsync()
...
-|[WhenAny: Bridging between the .NET Framework and the Windows Runtime](https://msdn.microsoft.com/library/jj635140(v=vs.120).aspx)|Shows how to bridge between Task types in the .NET Framework and IAsyncOperations in the [!INCLUDE[wrt](~/includes/wrt-md.md)] so that you can use <xref:System.Threading.Tasks.Task.WhenAny%2A> with a [!INCLUDE[wrt](~/includes/wrt-md.md)] method.|[Async Sample: Bridging between .NET and Windows Runtime (AsTask and WhenAny)](http://go.microsoft.com/fwlink/p/?LinkID=260638)|  
+|[WhenAny: Bridging between the .NET Framework and the Windows Runtime](https://msdn.microsoft.com/library/jj635140.aspx)|Shows how to bridge between Task types in the .NET Framework and IAsyncOperations in the [!INCLUDE[wrt](~/includes/wrt-md.md)] so that you can use <xref:System.Threading.Tasks.Task.WhenAny%2A> with a [!INCLUDE[wrt](~/includes/wrt-md.md)] method.|[Async Sample: Bridging between .NET and Windows Runtime (AsTask and WhenAny)](http://go.microsoft.com/fwlink/p/?LinkID=260638)|
```
Comment (mairaw) [inline docs/visual-basic/programming-guide/concepts/async/index.md]: ditto

**Review by mairaw — CHANGES_REQUESTED** [review]
LGTM. Left a few comments.

File: docs/csharp/language-reference/compiler-messages/cs2032.md [hunk 8]
```diff
@@ -47,10 +47,10 @@ Character 'character' is not allowed on the command-line or in response files
   
 2.  In the **My Documents** folder, create a text file that is named cs2032.cs and that contains whatever you want.  
   
-3.  Open the **Start** menu. Locate and select the [Developer Command Prompt for Visual Studio](https://msdn.microsoft.com/library/ms229859(v=vs.110).aspx).
+3.  Open the **Start** menu. Locate and select the [Developer Command Prompt for Visual Studio](https://docs.microsoft.com/dotnet/framework/tools/developer-command-prompt-for-vs).
```
Comment (GuardRex) [inline docs/csharp/language-reference/compiler-messages/cs2032.md]: @mairaw You want this change (use relative paths) made everywhere?

File: docs/framework/wpf/advanced/cleartype-registry-settings.md (hunk 4 above)
Comment (GuardRex) [inline docs/framework/wpf/advanced/cleartype-registry-settings.md]: It's the canonical URL, which is why the script dropped this in.

File: docs/standard/microservices-architecture/multi-container-microservice-net-applications/subscribe-events.md (hunk 5 above)
Comment (GuardRex) [inline docs/standard/microservices-architecture/multi-container-microservice-net-applications/subscribe-events.md]: Yes ... the anchor resulted in the link not getting picked up by the RegEx on this one.

File: docs/csharp/language-reference/compiler-messages/cs2032.md (hunk 8 above)
Comment (GuardRex) [inline docs/csharp/language-reference/compiler-messages/cs2032.md]: I think we should do this on another branch/PR ... I'm about to blow up GH here. :boom: ... 🚒 🚑 🚓

**Review by mairaw — APPROVED** [review]

**Comment by rpetrusha** [comment]
The build failed, but its status is not reported here, and the build report appears to be unavailable. Closing and reopening to start new build.

**Pull request merged**